source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -121,7 +120,7 @@ class ComponentInterface : NodeT {
    * map of predicates
    * @return the value of the predicate as a boolean
    */
-  [[nodiscard]] bool get_predicate(const std::string& predicate_name) const;
+  [[nodiscard]] bool get_predicate(const std::string& predicate_name);
```

Comment:
is there a reason this can't be const anymore? I guess because of the clock for the throttled logger? It's fine in that case

Reply:
yes exactly. This is actually annoying. I wonder if ROS found an alternative to that.

Reply:
I guess maybe you could instantiate a static clock inside the function. If it's scoped there and not in the class, the class could remain const. But it's not worth worrying about here for the time being